frontend/src/api/api.ts

```diff
@@ -499,6 +499,44 @@ export async function createCheckoutSession({ userId, priceId, successUrl, cance
     return session;
 }
 
+export async function getCustomerId({ subscriptionId }: { subscriptionId: any }): Promise<any> {
+    const response = await fetch("/get-customer", {
+        method: "POST",
+        headers: {
+           "Content-Type": "application/json" 
+        },
+        body: JSON.stringify({
+            subscription_id: subscriptionId
+        })
+    });
+    if (response.status > 299 || !response.ok) {
+        throw Error("Error creating checkout session")
+    }
+
+    const data = await response.json();
+    return data.customer_id;
+}
+
+export async function createCustomerPortalSession({ customerId, return_url }: any){
```

Comment:
@egdagger for createCustomerPortalSession, if the returned session has a known shape (e.g., contains url), you could define an interface

```node
interface BillingPortalSession {
  url: string;
  // other fields if needed
}

export async function createCustomerPortalSession({
  customerId,
  return_url
}: {
  customerId: string;
  return_url: string;
}): Promise<BillingPortalSession> {
    const response = await fetch("/create-customer-portal-session", {
        ...
    });
    // ...
    return await response.json() as BillingPortalSession;
}


```